client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -335,6 +349,54 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 		}
 	}
 
+	// ********** Logic for Premium Global Styles
+
+	const { shouldLimitGlobalStyles } = usePremiumGlobalStyles();
+	const [ showPremiumGlobalStylesModal, setShowPremiumGlobalStylesModal ] = useState( false );
+
+	function unlockPremiumGlobalStyles() {
+		// These conditions should be true at this point, but just in case...
+		if ( selectedDesign && selectedStyleVariation ) {
+			recordTracksEvent(
+				'calypso_signup_design_premium_global_styles_modal_show',
+				getEventPropsByDesign( selectedDesign, selectedStyleVariation )
+			);
+			setShowPremiumGlobalStylesModal( true );
+		}
+	}
+
+	function goToCheckoutForPremiumGlobalStyles() {
+		// These conditions should be true at this point, but just in case...
+		if ( selectedDesign && selectedStyleVariation && siteSlugOrId ) {
+			recordTracksEvent(
+				'calypso_signup_design_premium_global_styles_modal_checkout_button_click',
+				getEventPropsByDesign( selectedDesign, selectedStyleVariation )
+			);
+
+			const params = new URLSearchParams();
+
+			// When the user is done with checkout, send them back to the current url
+			const destUrl = new URL( window.location.href );
+			const destSearchP = destUrl.searchParams;
+
+			// Add &theme=theme_slug&style=style_slug to the query params
+			if ( selectedDesign.slug ) {
+				destSearchP.set( 'theme', selectedDesign.slug );
+				destSearchP.set( 'style', selectedStyleVariation.slug );
+				destUrl.search = destSearchP.toString();
+			}
+
+			const destString = destUrl.toString().replace( window.location.origin, '' );
```

Comment:
Nitpick (non-blocking): I think this other way is easier to mentally parse:

```js
const destString = destUrl.pathname + destUrl.search;
```

Reply:
I've simply copypasted from the regular checkout function, and TBH I'd rather keep it as-is (or replace both) so that they would both match a hypothetical search.